src/operator/numpy/np_elemwise_broadcast_op.cc

```diff
@@ -182,5 +114,211 @@ MXNET_OPERATOR_REGISTER_NP_BINARY_SCALAR(_npi_rpower_scalar)
+  [](const NodeAttrs& attrs) {                                                        \
+    return std::vector<std::pair<int, int> >{{0, 0}, {1, 0}};                         \
+  })                                                                                  \
+  .set_attr<FCompute>("FCompute<cpu>", TVMBinaryBroadcastCompute<func_##name##_cpu>)  \
+  .set_attr<nnvm::FGradient>("FGradient", MakeZeroGradNodes)                          \
+  .add_argument("lhs", "NDArray-or-Symbol", "First input to the function")            \
+  .add_argument("rhs", "NDArray-or-Symbol", "Second input to the function")
+
+MXNET_OPERATOR_REGISTER_NP_BINARY_LOGIC(equal);
+MXNET_OPERATOR_REGISTER_NP_BINARY_LOGIC(not_equal);
+MXNET_OPERATOR_REGISTER_NP_BINARY_LOGIC(greater);
+MXNET_OPERATOR_REGISTER_NP_BINARY_LOGIC(less);
+MXNET_OPERATOR_REGISTER_NP_BINARY_LOGIC(greater_equal);
+MXNET_OPERATOR_REGISTER_NP_BINARY_LOGIC(less_equal);
+
+#if MXNET_USE_CUDA
+#define MXNET_OPERATOR_REGISTER_NP_BINARY_LOGIC_GPU(name)                          \
+  NNVM_REGISTER_OP(_npi_##name)                                                    \
+  .set_attr<FCompute>("FCompute<gpu>", TVMBinaryBroadcastCompute<func_##name##_gpu>)
+
+MXNET_OPERATOR_REGISTER_NP_BINARY_LOGIC_GPU(equal);
+MXNET_OPERATOR_REGISTER_NP_BINARY_LOGIC_GPU(not_equal);
+MXNET_OPERATOR_REGISTER_NP_BINARY_LOGIC_GPU(greater);
+MXNET_OPERATOR_REGISTER_NP_BINARY_LOGIC_GPU(less);
+MXNET_OPERATOR_REGISTER_NP_BINARY_LOGIC_GPU(greater_equal);
+MXNET_OPERATOR_REGISTER_NP_BINARY_LOGIC_GPU(less_equal);
+#endif  // MXNET_USE_CUDA
+
+bool NumpyBinaryScalarLogicOpType(const nnvm::NodeAttrs& attrs,
+                                  std::vector<int>* in_attrs,
+                                  std::vector<int>* out_attrs) {
+  CHECK_EQ(in_attrs->size(), 1U);
+  CHECK_EQ(out_attrs->size(), 1U);
+  if (in_attrs->at(0) == -1) return false;
+  TYPE_ASSIGN_CHECK(*out_attrs, 0, mshadow::kBool);
+  return true;
+}
+
+template<const char* func>
+void TVMBinaryBroadcastScalarCompute(const nnvm::NodeAttrs& attrs,
+                                     const mxnet::OpContext& ctx,
+                                     const std::vector<TBlob>& inputs,
+                                     const std::vector<OpReqType>& req,
+                                     const std::vector<TBlob>& outputs) {
+  CHECK_EQ(inputs.size(), 1U);
+  CHECK_EQ(outputs.size(), 1U);
+  if (outputs[0].shape_.Size() == 0U) return;  // skip zero-size tensor
+
+  // prepare tblobs and TVMArgs
+  std::vector<TBlob> tblobs = {inputs[0], outputs[0]};
+  std::vector<int> type_codes;
+  std::vector<TVMValue> values;
+
+  const size_t num_args = 3;  // one input tensor, one scalar param, and one output
+  type_codes.resize(num_args);
+  values.resize(num_args);
+
+  // input tensor setup
+  type_codes[0] = kArrayHandle;
+  values[0].v_handle = const_cast<DLTensor*>(&(tblobs[0].dltensor()));
+
+  // scalar param
+  type_codes[1] = kDLFloat;
```

Comment:
@yzhliu Since I need to pass a `double` param to the op func generated by TVM, I cannot use the `Call` function defined in the `TVMOpModule`. I moved the logic of preparing `TVMArgs` up here from the `Call` function to MXNet op's `FCompute` function and added an independent `CallEx` in `TVMOpModule` to just invoke the kernel. We can discuss the change of the API to cater for more use cases.